File: ecommerce-project/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer!: Customer;  // '!' işareti ile bu değişkenin kesinlikle atanmış olduğunu belirtiyoruz
    shippingAddress!: Address;
    billingAddress!: Address;
    order!: Order;
    orderItems!: OrderItem[];
}

File: ecommerce-project/spring-boot-ecommerce/src/main/java/com/fathens/springbootecommerce/dto/Purchase.java
package com.fathens.springbootecommerce.dto;

import com.fathens.springbootecommerce.entity.Address;
import com.fathens.springbootecommerce.entity.Customer;
import com.fathens.springbootecommerce.entity.Order;
import com.fathens.springbootecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
